### PR TITLE
TOOLS-3252: Test support for Columnstore Indexes

### PR DESCRIPTION
**Pull request by tdq45gj**
This PR adds new integration tests for Columnstore Indexes in `mongodump` and `mongorestore`. 

In `mongodump`:
- Create collections with a Columnstore Index created with `columnstoreProjection` key.
- Dump the collections.
- Check the dumped meta files for the correct number of Columnstore Indexes, Columnstore Index name, and `columnstoreProjection` key.

In `mongorestore`:
- Create a collection with a Columnstore Index created with `columnstoreProjection` key.
- Dump the collection and restore it. 
- Check the restored collection for the correct number of Columnstore Indexes, Columnstore Index name, and `columnstoreProjection` key.

### CODE REVIEW

File: mongodump/mongodump_test.go [hunk 1]
```diff
@@ -1977,3 +2004,83 @@ func TestFailDuringResharding(t *testing.T) {
...
+
+	c2, err := countMetaDataFiles(dumpDBDir)
+	require.NoError(t, err)
+	require.Equal(t, c1, c2)
+
+	metaFiles, err := getMatchingFiles(dumpDBDir, ".*\\.metadata\\.json")
+	require.NoError(t, err)
+	require.Greater(t, len(metaFiles), 0)
+
+	oneMetaFile, err := os.Open(util.ToUniversalPath(filepath.Join(dumpDBDir, metaFiles[0])))
```
Comment (rohshar) [inline mongodump/mongodump_test.go:2058]: Would it be difficult to have it check the columnstore indexes for all relevant collections?

File: mongorestore/mongorestore_test.go [hunk 2]
```diff
@@ -2061,8 +2061,9 @@ func TestRestoreTimeseriesCollections(t *testing.T) {
 // ----------------------------------------------------------------------
 
 type indexInfo struct {
-	name string
-	keys []string
+	name                  string
+	keys                  []string
+	columnstoreProjection map[string]int32
```
Comment (rohshar) [inline mongorestore/mongorestore_test.go:2066]: Add a comment here

File: mongorestore/mongorestore_test.go [hunk 3]
```diff
@@ -2345,3 +2346,223 @@ func runMongodump(t *testing.T, dir, db, collection string, args ...string) stri
...
+	})
+
+	res := session.Database("admin").RunCommand(context.Background(), bson.M{"replSetGetStatus": 1})
+	if res.Err() != nil {
+		t.Skip("server is not part of a replicaset so we cannot test restore from oplog")
+	}
+	t.Run("restore from oplog", func(t *testing.T) {
+		testRestoreColumnstoreIndexFromOplog(t)
+	})
+	//t.Run("restore from oplog with default index name", func(t *testing.T) {
```
Comment (rohshar) [inline mongorestore/mongorestore_test.go:2374]: Did you mean to remove this?

File: mongorestore/mongorestore_test.go [hunk 4]
```diff
@@ -2345,3 +2346,223 @@ func runMongodump(t *testing.T, dir, db, collection string, args ...string) stri
...
+	//	testRestoreColumnstoreIndexFromOplog(t, "custom index name")
+	//})
+}
+
+func testRestoreColumnstoreIndexFromDump(t *testing.T) {
```
Comment (rohshar) [inline mongorestore/mongorestore_test.go:2379]: I know there aren't comments on the old code which is where the structure of these functions and the test as a whole are from, but I think its important that we add robust comments here. Lets add docstrings to all functions you add, so its very easy to follow along with what is happening at each point.

**Review by rohshar — COMMENTED** [review]
A few comments on clarity. Lets add a PR description.

File: mongodump/mongodump_test.go [hunk 5]
```diff
@@ -348,6 +348,33 @@ func setUpDBView(dbName string, colName string) error {
...
+	}
+
+	createIndexCmd := bson.D{
+		{"createIndexes", colName},
+		{"indexes", bson.A{
+			bson.D{
+				{"key", bson.D{{"$**", "columnstore"}}},
+				{"name", "dump_columnstore_test"},
+				{"columnstoreProjection", bson.D{
+					{`"`, 0},
```
Comment (autarch) [inline mongodump/mongodump_test.go:364]: Is this really legal syntax for this? That's really odd. Even if it's legal, can we instead do `{ "_id", 1 }` so it's less surprising?

File: mongorestore/mongorestore_test.go [hunk 6]
```diff
@@ -2063,6 +2063,8 @@ func TestRestoreTimeseriesCollections(t *testing.T) {
 type indexInfo struct {
 	name string
 	keys []string
+	// columnstoreProjection contains info of the columnstoreProjection key in columnstore indexes.
```
Comment (autarch) [inline mongorestore/mongorestore_test.go:2066]: nit: s/of the/about the/

File: mongorestore/mongorestore_test.go [hunk 7]
```diff
@@ -2345,3 +2347,228 @@ func runMongodump(t *testing.T, dir, db, collection string, args ...string) stri
...
+	testDB := session.Database(dbName)
+	defer func() {
+		err = testDB.Drop(nil)
+		if err != nil {
+			t.Fatalf("Failed to drop test database: %v", err)
+		}
+	}()
+
+	key := "$**"
+	columnstoreProjection := map[string]int32{"price": 0}
```
Comment (autarch) [inline mongorestore/mongorestore_test.go:2395]: While `: 0` apparently works, all the doc examples I've seen for indexes use `: 1` so let's use that instead.

File: mongorestore/mongorestore_test.go [hunk 8]
```diff
@@ -2345,3 +2347,228 @@ func runMongodump(t *testing.T, dir, db, collection string, args ...string) stri
...
```
Comment (autarch) [inline mongorestore/mongorestore_test.go:2555]: Let's use `: 1` here as well (and anywhere else I've missed).

**Review by autarch — APPROVED** [review]
LGTM % a few small things + Rohan's comments. No need to request another review once my comments are addressed.

File: mongodump/mongodump_test.go (hunk 1 above)
Comment (tdq45gj) [reply mongodump/mongodump_test.go]: No, just updated the test to check all collections.

**Pull request merged**